contrib/babelfishpg_tsql/src/catalog.c

```diff
@@ -2648,6 +2661,115 @@ rename_update_bbf_catalog(RenameStmt *stmt)
 	}
 }
 
+/*
+ * rename_object_update_bbf_schema_permission_catalog
+ *
+ * In case of renaming a table, view, procedure and function, modify the 'object_name' in
+ * 'babelfish_schema_permissions' to have consistency with the new names.
+ */
+static void
+rename_object_update_bbf_schema_permission_catalog(RenameStmt *stmt, int rename_type)
+{
+	/* Update 'object_name' in 'babelfish_schema_permissions' */
+	Relation	bbf_schema_rel;
+	TupleDesc	bbf_schema_dsc;
+	ScanKeyData key[4];
+	HeapTuple	tuple_bbf_schema;
+	HeapTuple	new_tuple;
+	SysScanDesc scan;
+	Datum		new_record_bbf_schema[BBF_SCHEMA_PERMS_NUM_OF_COLS] = {0};
+	bool		new_record_nulls_bbf_schema[BBF_SCHEMA_PERMS_NUM_OF_COLS] = {false};
+	bool		new_record_repl_bbf_schema[BBF_SCHEMA_PERMS_NUM_OF_COLS] = {false};
+	char		*logical_schema_name = NULL;
+	char		*object_name = NULL;
+	char		*object_type = NULL;
+	int16		dbid = get_cur_db_id();
+	ObjectWithArgs *objwargs;
+
+	/* open the catalog table */
+	bbf_schema_rel = table_open(get_bbf_schema_perms_oid(), RowExclusiveLock);
+	/* get the description of the table */
+	bbf_schema_dsc = RelationGetDescr(bbf_schema_rel);
+
+	if (rename_type == OBJECT_TABLE || rename_type == OBJECT_VIEW)
+	{
+		logical_schema_name = (char *) get_logical_schema_name(stmt->relation->schemaname, true);
+		object_name = stmt->relation->relname;
+		object_type = OBJ_RELATION;
+	}
+	else
+	{
+		if (rename_type == OBJECT_PROCEDURE)
+			object_type = OBJ_PROCEDURE;
+		else if (rename_type == OBJECT_FUNCTION)
+			object_type = OBJ_FUNCTION;
+		objwargs = (ObjectWithArgs *) stmt->object;
+		DeconstructQualifiedName(objwargs->objname, &logical_schema_name, &object_name);
+	}
+
+	/* search for the row for update => build the key */
+	ScanKeyInit(&key[0],
+				Anum_bbf_schema_perms_dbid,
+				BTEqualStrategyNumber, F_INT2EQ,
+				Int16GetDatum(dbid));
+	ScanKeyEntryInitialize(&key[1], 0,
+				Anum_bbf_schema_perms_schema_name,
+				BTEqualStrategyNumber, InvalidOid,
+				tsql_get_server_collation_oid_internal(false),
+				F_TEXTEQ, CStringGetTextDatum(logical_schema_name));
+	ScanKeyEntryInitialize(&key[2], 0,
+				Anum_bbf_schema_perms_object_name,
+				BTEqualStrategyNumber, InvalidOid,
+				tsql_get_server_collation_oid_internal(false),
+				F_TEXTEQ, CStringGetTextDatum(object_name));
+	ScanKeyEntryInitialize(&key[3], 0,
+				Anum_bbf_schema_perms_object_type,
+				BTEqualStrategyNumber,
+				InvalidOid,
+				tsql_get_server_collation_oid_internal(false),
+				F_TEXTEQ,
+				CStringGetTextDatum(object_type));
+
+	/* scan */
+	scan = systable_beginscan(bbf_schema_rel,
+			get_bbf_schema_perms_idx_oid(),
+			true, NULL, 4, key);
```

Comment:
@shalinilohia50 I gave this comment multiple times in your last PR, we are using index to scan but the key doesnt match all of the index columns

Reply:
Based on these 4 keys, there can be multiple rows depending on grantees. At this point, we don't know the grantee names. 
What do you suggest here?

Reply:
In fact, sp_rename doesn't care about grantee name IIUC. We need to modify all the rows that match the 4 keys irrespective of the grantee name.

Reply:
I think you didnt understand the issue
you are using index of 5 keys' oid to do a scan with 4 keys. Why are we not doing a scan on table's oid?
You are explaining why we need 4 keys and thats not an issue

Reply:
Updated.